fe_utils/mesh.py

```diff
@@ -110,8 +112,14 @@ def jacobian(self, c):
         :param c: The number of the cell for which to return the Jacobian.
         :result: The Jacobian for cell ``c``.
         """
+        cg1 = LagrangeElement(self.cell, 1)
+        # gradient degree 1 Lagrange basis
+        cg1fs = FunctionSpace(self, cg1)
```

Comment:
Careful! This builds a new `FunctionSpace` every time you construct a Jacobian. Since the construction of the global numbering is `O(n)` where n is the number of cells in the mesh, this makes Jacobian calculation `O(n)` instead of `O(1)`, and finite element assembly as a whole `O(n^2)`. This will make your code impractically slow.